File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        Map<String, Integer> map = new HashMap<>();
        for(String pa: participant) {
            map.computeIfAbsent(pa, k->0);
            map.put(pa, map.getOrDefault(pa, 0)+1);
        }
        for(String co: completion) {
            map.put(co, map.get(co)-1);
        }
        for(Map.Entry<String, Integer> entry: map.entrySet()){
            if(entry.getValue() != 0) {
                answer = entry.getKey();
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>
#include <iostream>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    
    map<string, int> ma;
    
    for(string name : participant) {
        ma[name]++;
    }

    for(string name : completion) {
        ma[name]--;
    }
    
    for(auto it : ma) {
        if(it.second > 0) {
            return it.first;
        }
    }
}